onnxruntime/contrib_ops/cuda/bert/ngram_repeat_block_impl.cu

```diff
@@ -20,35 +20,41 @@ using namespace onnxruntime::cuda;
 __global__ void banRepeatedTokens(const int64_t* __restrict__ tokens,
                                   float* __restrict__ lprobs,
                                   int max_predict_len, int vocab_size,
+                                  int recency_length,
                                   int no_repeat_ngram_size) {
   auto row = blockIdx.x;
   auto col = threadIdx.x;
   auto start = row * (max_predict_len) + col;
-  // Each thread compares ngram starting from
-  // thread index with final ngram starting from
-  // step - no_repeat_ngram_size +2
-  auto check_start_pos = blockDim.x;
-  auto lprob_start = row * vocab_size;
-  bool is_banned = true;
-  extern __shared__ int64_t tokens_shm[];
-  tokens_shm[col] = tokens[start];
-  if (col == blockDim.x - 1) {
-    for (int i = 1; i < no_repeat_ngram_size; i++) {
-      if (col + i < max_predict_len) {
-        tokens_shm[col + i] = tokens[start + i];
+  auto cur_len = (row + 1) * (max_predict_len);
```

Comment:
I'm uncertain about this `cur_len`. (based on previous code, it appears `cur_len == max_predict_length`). I think what you may want is to check if `col` (the token idx being checked in this thread) lies inside the `recency_length`. So the condition would be `recency_length > 0 && cur_len > recency_length && col >= (cur_len - recency_length)`.